src/js/utils/index.d.ts

```diff
@@ -34,6 +34,19 @@ declare const removeUndefined: <T extends object>(obj: T) => NonUndefinedProps<T
 
 export {isObject, deepFreeze, deepMerge, removeUndefined};
 
+/*
+ * Utility type for inferring the props type of a component.
+ *
+ * Example:
+ *
+ * ```typescript
+ * import { Box } from 'grommet';
+ *
+ * type BoxProps = PropsOf<typeof Box>;
```

Comment:
Good point! Let me see if I can adjust the example to be more clear here.

One thing I've been wondering is if ultimately the props types that Grommet provides (like `BoxProps`) should be updated to include all of the props including intrinsic props (typically this is what I'm used to in my own projects). I'm not sure if there was a historical reason why it was done that way nor am I sure if it would be a breaking change for some TS users who rely on it that way though. Probably outside the scope of this PR but could be worth considering in another one.